packages/ruleset/src/functions/delete-body.js

```diff
@@ -12,6 +12,22 @@ module.exports = function (operation, _opts, context) {
   if (!logger) {
     ruleId = context.rule.name;
     logger = LoggerFactory.getInstance().getLogger(ruleId);
+
+    // This rule has been deprecated and is disabled by default (i.e. severity set to 'off'),
+    // but spectral will still invoke the rule despite the severity setting.
+    // If a user enables this rule in their custom ruleset, we'll want to log a warning
+    // to alert them to the deprecation status.
+    if (context.rule && context.rule.definition) {
+      const severity = context.rule.definition.severity;
+      if (
+        (typeof severity === 'number' && severity >= 0) ||
+        (typeof severity === 'string' && severity !== 'off')
+      ) {
+        logger.warn(
```

Comment:
Will this log more than once if the rule is enabled and there are multiple operations?

Reply:
No, this is inside the "if (!logger)" above, which we'll enter only the first time we are invoked.  I initially started setting up a separate flag to make sure we log the warning only once, then realized I could piggyback off the if block that was already there to initialize the logger.

Reply:
Ah, of course - I figured you wouldn't have implemented it in a way that would log multiple messages but I was struggling to understand how you did it (totally missed that context 🙂)